pysonde/thermodynamics.py

```diff
@@ -46,93 +48,141 @@ def convert_rh_to_dewpoint(temperature, rh):
+
     if method == "hardy1998":
-        g = np.empty(8)
-        g[0] = -2.8365744 * 10**3
-        g[1] = -6.028076559 * 10**3
-        g[2] = 1.954263612 * 10**1
-        g[3] = -2.737830188 * 10 ** (-2)
-        g[4] = 1.6261698 * 10 ** (-5)
-        g[5] = 7.0229056 * 10 ** (-10)
-        g[6] = -1.8680009 * 10 ** (-13)
-        g[7] = 2.7150305
-
-        e_sw = np.zeros_like(temperature_K)
-
-        for t, temp in enumerate(temperature_K):
-            ln_e_sw = np.sum([g[i] * temp ** (i - 2) for i in range(0, 7)]) + g[
-                7
-            ] * np.log(temp)
-            e_sw[t] = np.exp(ln_e_sw)
-        if isinstance(temperature, pp.pint_array.PintArray):
-            pp.PintType.ureg = ureg
-            e_sw = pp.PintArray(e_sw, dtype="Pa")
-        elif isinstance(temperature, xr.core.dataarray.DataArray) and hasattr(
-            temperature.data, "_units"
-        ):
-            e_sw = xr.DataArray(
-                e_sw, dims=temperature.dims, coords=temperature.coords
-            ) * metpy.units.units("Pa")
-        return e_sw
+        # print("I use hardy1998 for e_s!!!")
+        g = np.array(
+            [
+                -2.8365744e3,
+                -6.028076559e3,
+                1.954263612e1,
+                -2.737830188e-2,
+                1.6261698e-5,
+                7.0229056e-10,
+                -1.8680009e-13,
+                2.7150305,
+            ]
+        )
+
+        e_sw = np.exp(
+            np.sum([g[i] * temperature_K ** (i - 2) for i in range(7)], axis=0)
+            + g[7] * np.log(temperature_K)
+        )
+
+    elif method == "wagner_pruss":
+        # print("I use wagner_pruss for e_s!!!")
+        e_sw = mtsvp.liq_wagner_pruss(temperature_K)
+
+    else:
+        raise ValueError(
+            f"Unknown method '{method}'. Available methods: 'hardy1998', 'wagner_pruss'."
+        )
+
+    # Convert result back to PintArray or xarray DataArray
+    if isinstance(temperature, pp.pint_array.PintArray):
+        pp.PintType.ureg = ureg
+        e_sw = pp.PintArray(e_sw, dtype="Pa")
+    elif isinstance(temperature, xr.DataArray) and hasattr(temperature.data, "_units"):
+        e_sw = xr.DataArray(
+            e_sw, dims=temperature.dims, coords=temperature.coords
+        ) * metpy.units.units("Pa")
+
+    return e_sw
 
 
 def calc_wv_mixing_ratio(sounding, vapor_pressure):
     """
-    Calculate water vapor mixing ratio
+    Works for both LEVEL-1 and LEVEL-2.
+
+    Key improvements:
+    - Handles both raw numeric values and Pint-quantified inputs.
+    - Dynamically detects pressure variable ("p" or "pressure").
+    - Ensures unit consistency across different dataset formats.
+    - Includes error handling for missing or incompatible units.
+
+    Parameters:
+    ----------
+    sounding : xarray.Dataset
+        Dataset containing pressure values.
+    vapor_pressure : xarray.DataArray or pint.Quantity
+        Vapor pressure values (must have units of Pascal).
```

Comment:
I think any pressure unit can be used as long it is compatible with `pint.to("Pa")`

```suggestion
        Vapor pressure values
```